sklearn/preprocessing/imputation.py

```diff
@@ -157,6 +158,14 @@ def fit(self, X, y=None):
             X = check_array(X, accept_sparse='csc', dtype=np.float64,
                             force_all_finite=False)
 
+            # Replace nan columns with the passed `default_value`.
+            if self.default_value is not None:
+                _isnan = np.vectorize(Imputer._isnan)
+                is_col_nan = np.all(_isnan(X), axis=0)
+                if np.any(is_col_nan):
+                    nan_indexes = np.where(is_col_nan)[0]
+                    X[:, nan_indexes] = self.default_value
```

Comment:
You're modifying the input! How about we just modify the calculated statistics below?

Reply:
That's a good hint, because the data structure of the calculated statistics is simpler.